### PR TITLE
`Development`: Remove unused variables, fields, and empty super constructor calls

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Description

Removes unused variables, fields, and empty super constructor calls.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged

## Summary by CodeRabbit

- **New Features**
	- Enhanced submission handling with version management.
	- Improved post creation with extended associations for Exercise, Lecture, and Course.
	- Advanced quiz functionalities with new methods for Drag and Drop and Short Answer questions.
- **Refactor**
	- Simplified constructors across various classes for better compatibility and readability.
	- Streamlined ErrorType enumeration for consistency error reporting.
	- Updated conversation messaging service by refining dependencies and improving notification management.
- **Tests**
	- Updated integration and service tests with refined mock setups and simplified domain entity declarations.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/domain/Submission.java [hunk 1]
```diff
@@ -257,6 +281,14 @@ public void setParticipation(Participation participation) {
...
     }
 
+    public Set<SubmissionVersion> getVersions() {
+        return versions;
+    }
+
+    public void setVersions(Set<SubmissionVersion> submissionVersions) {
+        this.versions = submissionVersions;
+    }
+
```
Comment (rstief) [inline src/main/java/de/tum/in/www1/artemis/domain/Submission.java:291]: Is there a reason for adding getters/setters? If this is on purpose, it should probably be added to the PR description/title

**Review by rstief — APPROVED** [review]
Code LGTM, one small comment 👍

File: src/main/java/de/tum/in/www1/artemis/domain/Submission.java (hunk 1 above)
Comment (MaximilianAnzinger) [reply src/main/java/de/tum/in/www1/artemis/domain/Submission.java:291]: I added it to the description 👍

**Review by milljoniaer — APPROVED** [review]
Code looks good to me 🧹  🚀

**Comment by krusche** [comment]
Not sure if it makes sense to add unused getters and setters. This reduces the code coverage and does not really help to be honest

**Comment by MaximilianAnzinger** [comment]
@krusche I removed the getters and setters 👍

**Review by julian-christl — APPROVED** [review]
code

**Review by JohannesWt — APPROVED** [review]
Code looks good

**Review by Jan-Thurner — APPROVED** [review]
code LGTM

**Pull request merged**